eureka-client/src/main/java/com/netflix/discovery/converters/jackson/EurekaJsonJacksonCodec.java

```diff
@@ -37,7 +39,7 @@ public EurekaJsonJacksonCodec() {
     public EurekaJsonJacksonCodec(final KeyFormatter keyFormatter, boolean compact) {
```

Comment:
I can imaging potential future cases where we may have multiple different compact types. As this is exposed as public API, can we change this from a boolean to a more flexible type?